analysis-#60 montgomery county vacant lots + spatial join

pspauster opened the pull request:
Vacant Lots Analysis - Montgomery, (montgomery file) + Gaithersburg +Rockville (montgomery_YS file)

pspauster committed:
pulled down new branch with code i needed

pspauster committed:
wrote join code which should run. getting an error with the shapes in the underlying file.

pspauster committed:
clean spatial join :) only 0.18% didn't match with parcels in jur.

pspauster committed:
Split the spatial merge and the vacant lots analysis into two markdowns. The post clean will include the spatial merge and fill in missing values for lot size and zoning. The vacant lots analysis will process the zoning codes and run the vacant lots calculation

pspauster committed:
Cleaned and Merged both the parcel and zoning files to the cleaned BK data. Replaced missing values for lotsize and zoning code where possible.

pspauster committed:
Merged the zoning file and the parcel file together. Began writing the clean-montgomery script

pspauster committed:
cleaned and imported the zoning codes, created factors

pspauster committed:
prepped for Yi Peng to help

maggiesu0725 committed:
clean rockville and Gaithersburg, add comments for changes

pspauster committed:
need to clean zoning code from BK and then ready to calculate

pspauster committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

Yi Peng added a new file

pspauster committed:
completed preliminary calculation - need to clean table output to fit what leah asked for

maggiesu0725 committed:
clean zoning code by jurisdiction

maggiesu0725 committed:
merge with Mongomery code where Rockvile code is not available.

maggiesu0725 committed:
read in rockvile zoning layer

maggiesu0725 committed:
update rockvile flag

pspauster committed:
Added flags for rockville and gaithersburg.

pspauster committed:
Changed filter on vacant calculation

pspauster committed:
Finished vacant lots analysis, to combine with Rock and Gaith

maggiesu0725 committed:
test

maggiesu0725 committed:
Revert "test"

This reverts commit 28bcfb2c86f89ad2c34dafdd4e46ec7e78db41af.

pspauster committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60
"Merge remote into local"

maggiesu0725 committed:
calculate Rockville and Gaithersburg units

pspauster committed:
Update vacant-lots-montgomery.Rmd

Actually for real done with the analysis this time. Need to add some annotations

pspauster committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

maggiesu0725 committed:
apply vacant filter after the spatial join

maggiesu0725 committed:
update units calc

maggiesu0725 committed:
update rockville and gaithersburg calc

maggiesu0725 committed:
recategorize gaithersburg zoning code

maggiesu0725 committed:
slice twice the dataset for max density

pspauster committed:
finished vacant lots

finalized vacant lots calculation and created output for max units

pspauster committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

pspauster committed:
Update vacant-lots-montgomery.Rmd

table output

pspauster committed:
Update vacant-lots-montgomery.Rmd

one final thing for the table

pspauster commented:
@lhendey the postclean-montgomery, read_montgomery, and vacant-lots-montgomery are ready for your review

maggiesu0725 committed:
update MXD and PD calculation for Rockville and Gaithersburg

lhendey commented:
@pspauster @maggiesu0725 - How does vacant_lots_montgomery _YS relate to vacant_lots_montgomery?

lhendey commented:
@pspauster - i need some direction on the order of the files. Where does postclean and spatial join fit in? 

please add descriptions of each file (and how they connect together) to each header
update all header text (at least one says arlington).  

i can't knit postclean-montgomery, spatialjoin-montgomery, vacant-lots-montgomery (generally before requesting review - please knit all files).  

this is the error in all- (happens quickly) - let me know if there is some other package that i need to install. 

 Error in parse_block(g[-1], g[1], params.src) : duplicate label 'read'
Calls: <Anonymous> ... process_file -> split_file -> lapply -> FUN -> parse_block
Execution halted

my best review windows are limited today - I have from 10:30 -11 and 12-1.

pspauster commented:
@lhendey I'm on the changes now and will tag you when it's pushed. But to answer your first questions...

The order for montgomery is as follows - 
1)preclean
2)clean
3)postclean
**then it branches
4a)vacant-lots-montgomery (montgomery county minus cities)
4b)vacant-lots-montgomery_YS (rockville and gaithersburg)

The spatial join file is defunct - it became the post clean, so don't review it.

pspauster committed:
knit and small tweaks

Knitted

pspauster committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

lhendey commented on Prog/vacant-lots-montgomery_YS.Rmd:
````diff
@@ -0,0 +1,1257 @@
+```
+
+
+Look more closely at vacant lots < 50,000 SF
````
not just vacant, right?

lhendey commented on Prog/vacant-lots-montgomery_YS.Rmd:
```diff
@@ -0,0 +1,1257 @@
+                     mutate(unit_bk= ifelse(category=="sf", 1, numberofunits))
+
+
+Rockvillevacant <- zoningdata_rockvile1_final2 %>% 
```
does this incorporate the assessment values to select out vacant with structure?

maggiesu0725 replied:
yes

lhendey commented on Prog/vacant-lots-montgomery_YS.Rmd:
```diff
@@ -0,0 +1,1257 @@
+
+1. Buildable area: this is `lotsize_sf` x the coverage ratio (`lot_coverage_max`)
+2. Living area: this is the buildable area x the construction efficiency. The lower bound estimate is 70%, the upper bound estimate is 85%.
+3. Units per floor: this is the living area divided by the average unit size. We use 900 sq. feet, based on the American Housing Survey 2017 data for the DC metro area.
```
940

lhendey commented on Prog/vacant-lots-montgomery_YS.Rmd:
```diff
@@ -0,0 +1,1257 @@
+              units_height = sum(units_height, na.rm = TRUE),
+              units_lotsize = sum(units_lotsize, na.rm = TRUE),
+              units_total = sum(units_total, na.rm = TRUE),
+              units_per_lot = floor(sum(units_total/lots))) %>% 
```
so there aren't any reseting to 1 unit for single family like in some of the others? (for rockville too)?

maggiesu0725 replied:
updated one unit flag for gaithersburg.

lhendey commented on Prog/vacant-lots-montgomery_YS.Rmd:
```diff
@@ -0,0 +1,1257 @@
+                 construction_efficiency = .7,
+                 average_unit_size = 940,
+                 dwelling_type_breakdown = TRUE) %>% 
+  knitr::kable()
```
units per lot is coming out to 0 for SF?

maggiesu0725 replied:
less than one will be rounded to 0 in the summary table. I added a new variable that's the unrounded unit per lot.

lhendey commented on Prog/vacant-lots-montgomery_YS.Rmd:
```diff
@@ -0,0 +1,1257 @@
+  group_by(housing_type) %>% 
+  count() %>% 
+  knitr::kable()
+
```
units per lot coming out to 0 for SF?

maggiesu0725 replied:
less than one will be rounded to 0 in the summary table. I added a new variable that's the unrounded unit per lot.

lhendey reviewed (changes requested):
Thanks @pspauster @maggiesu0725 i have many comments embedded in this review - may be easier to look on the github website. generally things look pretty good - most comments are around better documentation of the code, i wasn't usually running the data and need to be understand WHY you are making certain decisions. i'd like to eliminate the duplication between the two vacant lots programs. please coordinate to do that. thanks!

pspauster commented:
@lhendey got it - I hear you on the documentation. Yi Peng, I'll figure out the best place to write out something that can be the start of the rock/gaith file. I do think it makes sense to keep them separate because the calculations/functions are different. My day is busy today but I'll be able to look more closely and add stuff tomorrow.

maggiesu0725 committed:
address most of Leah's comments

maggiesu0725 commented on Prog/vacant-lots-montgomery_YS.Rmd:
````diff
@@ -0,0 +1,1257 @@
+```{r list codes Rockville}
+zoning_rockville %>% 
+  count(dwelling_type) %>% 
+  knitr::kable()
````
The dwelling type information was missing from the zoning spreadsheet we have. But because there's no duplicate codes in the zoning spreadsheet, we don't really need it for creating the factor variable. I'm using the dwelling type information in the zoning ordinance to create categories for the table shells, though.

maggiesu0725 commented on Prog/vacant-lots-montgomery_YS.Rmd:
````diff
@@ -0,0 +1,1257 @@
+```
+
+### Create factor variable
+All rockville zoning codes are unique, so we do not need a factor. We will automatically select "all", or treat it as dwelling type is NA. In addition, rockville zoning code doesn't have dups except for MXTD and LPNC, which are for parks and mixes use transition. We don't have a density preference over the two. 
````
We are assigning order within each code so that we get a 1:1 merge, so we are not comparing the density between mixed use and parks.

maggiesu0725 commented on Prog/vacant-lots-montgomery_YS.Rmd:
````diff
@@ -0,0 +1,1257 @@
+```
+
+
+## Clean lots data
````
both

maggiesu0725 commented on Prog/vacant-lots-montgomery_YS.Rmd:
```diff
@@ -0,0 +1,1257 @@
+  count(zoning_code) %>% 
+  knitr::kable()
+
+#gaithershburg also has R200(5) RE2C(1) which should be a montegomery codes -- what to do? 
```
we spatial joined these with the Gaithersburg shapefile so these lots are definitely in Gaithersburg. So I haven't do any calculation on these lots because they don't have zoning parameters merged on.

maggiesu0725 committed:
add unrounded units per lot to summary table

pspauster commented on Prog/postclean-montgomery.Rmd:
```diff
@@ -0,0 +1,607 @@
+
+
+parcel_sf1 <- parcel_sf1 %>% 
+  filter(validgeo == 1 & !is.na(validgeo))
```
it should be printing the line above count(validgeo)

pspauster commented on Prog/postclean-montgomery.Rmd:
```diff
@@ -0,0 +1,607 @@
+
+paste0(sum(!is.na(pzjoin1$ZONE)), " observations did join- ",
+      round(sum(!is.na(pzjoin1$ZONE)) / nrow(pzjoin1) * 100, 2), "%")
+
```
this actually says "did join" so the 16,000 or so are the ones that ARE in rockville, which seems right to me

pspauster commented on Prog/postclean-montgomery.Rmd:
```diff
@@ -0,0 +1,607 @@
+
+pzjoin_df1 %>%  as.data.frame() %>% 
+  count(notinmont, rockflag, gaitflag)
+
```
that's the correct interpretation! those will get filtered out in the next file. These are the properties we think are in the various other cities with different codes

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+  Comments = col_character()
+)
+
+zoningcode <- read_csv(paste0("L:/Libraries/RegHsg/Doc/zoning-codes/",
```
Yeah, here it's filling in whatever it can from the mont/rockville zoning shape files.

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+                       col_types = z)
+
+names(zoningcode) <- 
+  c("juristdiction", "zoning_code", "zone", "dwelling_type",
```
I spelled it wrong consistently throughout. But it is spelled wrong the same way

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+4. duplex-side
+5. detached house
+
+multi-use and general have the same zoning specs as apartment. I change all of the multi-use and general zoning codes to apartments for simplicity.
```
This one might require another look - maybe Sarah has thoughts. As of now I don't think I make any allowances for mixed use in the calculation

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
````diff
@@ -0,0 +1,929 @@
+```
+
+
+I created a dataset with results for troubleshooting purposes. Then I checked to make sure there aren't more townhouses than are allowed to be built on any one property.
````
Yeah I don't need these anymore after i figured out what went wrong and fixed it. will delete

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
````diff
@@ -0,0 +1,929 @@
+```
+
+
+Here I did trouble shooting for the units that produced no units. ignore
````
Yeah I don't need these anymore after i figured out what went wrong and fixed it. will delete

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+zoning_montgomery <- zoningcode %>% 
+  filter(juristdiction == "Montgomery")
+
+zoning_rockville <- zoningcode %>% 
```
@maggiesu0725 added a csv write out if you want to use it - once I run it it will be rockville-zoning-code-clean.csv and gaithersburg-zoning-code-clean.csv in L:/Libraries/RegHsg/Doc/zoning-codes

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
````diff
@@ -0,0 +1,929 @@
+Count how many vacant lots
+```{r filtering}
+
+jur %>% 
````
added comment with reasoning. Also chris used this threshold in fairfax cleaning I believe

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+
+jur %>% 
+  filter(lotsize_sf>1000000) %>% 
+count(lotsize_sf, propaddress)
```
added a line to the post clean file that fills in with the county shape file parcel area if the lot size is above 1 million

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+montfinal <- mont %>% 
+  mutate(zoning_code = case_when(
+    zoning %in% c("CR1.0","CR0.5", "CR2.0", "CR5.0", "cR3", "CR5", "CR3.0", "CR4.0") ~ "CR",
+    zoning %in% c("RT","RT6", "RT6.0") ~ "RT-6.0",
```
Nope, all the other RTs (8, 10, 12 are cleaned and assigned to that code. The general RTs without a number went to 6 because that is the most restrictive zoning code

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
````diff
@@ -0,0 +1,929 @@
+
+```
+
+There are `r ` observations that did not have a match in the zoning code. Below, we see which zoning codes were in the Black Knight data but NOT in the zoning code data.
````
there is no output from the previous chunk so that will not work - removed it

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
````diff
@@ -0,0 +1,929 @@
+```{r antijoin}
+
+anti_join(montfinal, zoning_montgomery1, by = c("zoning_code")) %>% 
+  count(zoning) %>% 
````
Some of them are rockville and gaithersburg, the rest are non residential. added a note to the code

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+1. filter out non residential properties
+2. fill in lot area minimums from site minimums
+3. fill in lot coverage maximums from site coverage maximums
+4. fill in per unit....*patrick do this 3/20*
```
I did this and didn't remove the note for myself

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+           units_density = floor(density * lotsize_acres),
+           units_lotsize = floor(lotsize_sf / lotarea_per_unit),
+           units_per_floor = floor(living_area/average_unit_size),
+           units_FAR = units_per_floor * FAR_stories) %>% 
```
done

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+                one_unit_flag == 1 ~ 1,
+                two_unit_flag == 1 ~ 2,
+                TRUE ~ .))) %>% 
+    mutate(units_total = case_when(
```
I used a pmin function which just selects the lowest number of units from that list of variables for each lot, I think @maggiesu0725  wrote different logic that does the same thing.

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+                 average_unit_size = 940,
+                 dwelling_type_breakdown = TRUE) %>% 
+  knitr::kable()
+
```
@lhendey let's talk about this one Monday, I'm not sure if I'm understanding it correctly

pspauster commented on Prog/vacant-lots-montgomery.Rmd:
```diff
@@ -0,0 +1,929 @@
+                 construction_efficiency = .7,
+                 average_unit_size = 940,
+                 dwelling_type_breakdown = TRUE) %>% 
+  knitr::kable()
```
@lhendey this one on monday too

pspauster replied:
@lhendey closing this loop. The new output has some FAR units for apartment and none for townhouse. That is due to the fact that the only zoning code with FAR restrictions are the mixed use zoning codes. All of these are apartments or they are not large enough to build on.

pspauster committed:
most of the changes from leah's comments

pspauster commented on Prog/postclean-montgomery.Rmd:
````diff
@@ -0,0 +1,607 @@
+```
+
+----------------------
+Set joined parcel as SF to faciltate join with zoning layer
````
figured this one out - formatting error at the top

pspauster committed:
more edits

maggiesu0725 committed:
udpate zoning spreadsheet sources

pspauster committed:
updated far calculation, adjusted formatting.

should be just about done

pspauster committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

pspauster committed:
last commit (hopefully) for report

lhendey commented:
@maggiesu0725 in gaithersburg vacant lots output units_total=0 for all types.

maggiesu0725 committed:
minor fix and knit

lhendey committed:
begin code to fix lots

lhendey committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

sstrochak committed:
fix spatial join

sstrochak committed:
re-knit mont

maggiesu0725 committed:
update lot coverage to 0.75 when NA

Commit:
cleaning mongtomery postclean

lhendey committed:
Update postclean-montgomery.Rmd

sstrochak committed:
use property and address file

sstrochak committed:
major restructure, remove parcel join

sstrochak committed:
knit

sstrochak committed:
cosmetic tweak

sstrochak committed:
recategorize land use based on county property file

sstrochak committed:
add back vars from property file

sstrochak committed:
check category before export

sstrochak committed:
update with new data

sstrochak committed:
update with new data

lhendey committed:
fix to lot coverage max

@sstrochak - does this fix look right to you? output looks better.

lhendey committed:
update zoning read in.

lhendey committed:
reknit

sstrochak committed:
add density calculation

lhendey committed:
Update postclean-montgomery.Rmd

@sstrochak - can you look at my changes?  i'm knitting the file now.  a few obs got added at one point - i wanted you to look at.

lhendey committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

sstrochak committed:
find duplicates in rockvillw join

lhendey committed:
Update vacant-lots-montgomery.Rmd

lhendey committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60

sstrochak committed:
try and fix duplicates

sstrochak committed:
remove overlap in zoning layer

sstrochak committed:
drop extra vars and knit

sstrochak committed:
Merge branch 'analysis-#60' of https://github.com/NeighborhoodInfoDC/RegHsg into analysis-#60
i update

sstrochak committed:
update comments

sstrochak committed:
change write out

sstrochak committed:
update comments and re-knit final documents

sstrochak committed:
rename file, add table results, fix formatting, update comments

sstrochak committed:
knit montgomery to check consistency

sstrochak committed:
knit to check rockville/gaith consistency

sstrochak committed:
Merge branch 'master' into analysis-#60
